src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/UploadEnvelopeDocumentsService.java

```diff
@@ -113,8 +109,8 @@ private void processEnvelope(CloudBlobContainer blobContainer, Envelope envelope
                 exception
             );
         } finally {
-            if (blobClient != null) {
-                tryBreakLease(blobClient);
+            if (leaseId.isPresent()) {
+                blobManager.tryReleaseLease(blob, containerName, zipFileName, leaseId.get());
```

Comment:
This is because the task should release its own lease instead of breaking any lease. It's most likely its own lease, but that's not guaranteed.